### PR TITLE
Sdk 172

### PR DESCRIPTION
**Pull request by ZahidZafar**
Add checks so that only "String", "integer" "float"/"double" and "boolean" values are accepted.

### CODE REVIEW

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Remove the unrelated parts

**Comment by ArtursKadikis** [comment]
Add a test function in the sample app to verify that this works correctly.

File: Assets/Example/Scripts/CountlyEntryPoint.cs [hunk 1]
```diff
@@ -393,3 +408,15 @@ public void OnNotificationClicked(string message, int index)
         Debug.Log("[Countly Example] OnNoticicationClicked: " + message + ", index: " + index);
     }
 }
+
+public class CustomDataType
```
Comment (ArtursKadikis) [inline Assets/Example/Scripts/CountlyEntryPoint.cs:412]: No need to create a custom data type for this. Just use any already available one

File: Assets/Example/Scripts/CountlyEntryPoint.cs [hunk 2]
```diff
@@ -221,6 +221,21 @@ public async void EventWithSumAndSegmentation()
        
     }
 
+    public async void EventWithInvalidSegmentation()
+    {
+        long currentMillis = System.DateTime.UtcNow.Millisecond; // invalid data type
+        CustomDataType customDataType = new CustomDataType("foo", 49); // invalid data type
+        SegmentModel segment = new SegmentModel(new Dictionary<string, object>
+        {
+            { "Current Millis", currentMillis},
```
Comment (ArtursKadikis) [inline Assets/Example/Scripts/CountlyEntryPoint.cs:230]: To make this tests useful, add all accepted data types and a couple of types that should be rejected (2 is fine).

File: Assets/Example/Scenes/EntryPoint.unity [hunk 3]
```diff
@@ -231,7 +231,7 @@ RectTransform:
   m_Children:
   - {fileID: 1923053171}
   m_Father: {fileID: 2006651435}
-  m_RootOrder: 19
```
Comment (ArtursKadikis) [inline Assets/Example/Scenes/EntryPoint.unity:234]: Why does this scene file get changed?

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

File: Assets/Example/Scenes/EntryPoint.unity (hunk 3 above)
Comment (ZahidZafar) [reply Assets/Example/Scenes/EntryPoint.unity:234]: Ui added for testing invalid segmentation.

Comment (ArtursKadikis) [reply Assets/Example/Scenes/EntryPoint.unity:234]: Share a screenshot of that UI

File: Assets/Example/Scripts/CountlyEntryPoint.cs (hunk 1 above)
Comment (ZahidZafar) [reply Assets/Example/Scripts/CountlyEntryPoint.cs:412]: then i am going to use counlty object.

File: Assets/Example/Scripts/CountlyEntryPoint.cs [hunk 4]
```diff
@@ -18,8 +18,8 @@ private void Awake()
     {
         CountlyConfiguration configuration = new CountlyConfiguration
         {
-            ServerUrl = "https://try.count.ly/",
-            AppKey = "YOUR_APP_KEY",
+            ServerUrl = "https://master.count.ly/",
```
Comment (ArtursKadikis) [inline Assets/Example/Scripts/CountlyEntryPoint.cs:21]: garbage

File: Assets/Example/Scenes/EntryPoint.unity (hunk 3 above)
Comment (ArtursKadikis) [reply Assets/Example/Scenes/EntryPoint.unity:234]: still not uploaded

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

**Comment by ZahidZafar** [comment]
Screenshot 2020-12-03 at 7 26 59 PM

**Comment by ArtursKadikis** [comment]
Added a new section to the UI called "SDK tests"
Move that new button into it. Add also the "Events With Multithreads" button there and rename it to "Events with multiple threads"

**Comment by ArtursKadikis** [comment]
Afterwards send a screenshot of how it looks

**Comment by ZahidZafar** [comment]
> Afterwards send a screenshot of how it looks

Screenshot 2020-12-03 at 8 06 18 PM

**Comment by ZahidZafar** [comment]
> Afterwards send a screenshot of how it looks

**Comment by ZahidZafar** [comment]
Screenshot 2020-12-03 at 8 20 45 PM

**Review by ArtursKadikis — APPROVED** [review]

**Pull request merged**